octodns/provider/easydns.py

```diff
@@ -0,0 +1,450 @@
+                             .format(self.auth_key)})
+        sess.headers.update({'accept': 'application/json'})
+        self._sess = sess
+
+    def _request(self, method, path, params=None, data=None):
+        url = '{}{}'.format(self.base_path, path)
+        resp = self._sess.request(method, url, params=params, json=data)
+        if resp.status_code == 400:
+            self.log.debug('Response code 400, path=%s', path)
+            if method == 'GET' and path[:8] == '/domain/':
+                raise EasyDNSClientNotFound()
+            raise EasyDNSClientBadRequest()
+        if resp.status_code == 401:
+            raise EasyDNSClientUnauthorized()
+        if resp.status_code == 403 or resp.status_code == 404:
+            raise EasyDNSClientNotFound()
+        resp.raise_for_status()
+        return resp
+
+    def domain(self, name):
+        path = '/domain/{}'.format(name)
+        return self._request('GET', path).json()
+
+    def domain_create(self, name):
+        # EasyDNS allows for new domains to be created for the purpose of DNS
+        # only, or with domain registration. This function creates a DNS only
+        # record expectig the domain to be registered already
+        path = '/domains/add/{}'.format(name)
+        domain_data = {'service': 'dns',
+                       'term': 1,
+                       'dns_only': 1,
+                       'portfolio': self.domain_portfolio,
+                       'currency': self.default_currency}
+        self._request('PUT', path, data=domain_data).json()
+
+        # EasyDNS creates default records for MX, A and CNAME for new domains,
+        # we need to delete those default record so we can sync with the source
+        # records, first we'll sleep for a second before gathering new records
+        # We also create default NS records, but they won't be deleted
+        sleep(1)
+        records = self.records(name, True)
+        for record in records:
+            if record['host'] in ('', 'www') \
+               and record['type'] in ('A', 'MX', 'CNAME'):
+                self.record_delete(name, record['id'])
+
+    def records(self, zone_name, raw=False):
+        if raw:
+            path = '/zones/records/all/{}'.format(zone_name)
+        else:
+            path = '/zones/records/parsed/{}'.format(zone_name)
+
+        ret = []
+        resp = self._request('GET', path).json()
+        ret += resp['data']
+
+        for record in ret:
+            # change any apex record to empty string
+            if record['host'] == '@':
+                record['host'] = ''
+
+            # change any apex value to zone name
+            if record['rdata'] == '@':
+                record['rdata'] = '{}.'.format(zone_name)
+
+            # EasyDNS supports DYNamic A records so we'll convert these
+            # to their underlying DNS record type before processing
+            if record['type'] == "DYN":
+                record['type'] = 'A'
```

Comment:
Sorry didn't notice this one under the URL and STEALTH types before. What's the behavior of `DYN` records? Will it have similar problems to STEALTH and URL in terms of losing information and being impossible to manage?

Reply:
Sorry I overlooked that too, it's been removed, just pushed the update.